File: app/src/main/java/com/faendir/lightning_launcher/multitool/proxy/PropertyEditor.java
package com.faendir.lightning_launcher.multitool.proxy;

/**
 * @author lukas
 * @since 04.07.18
 */
public interface PropertyEditor extends Proxy {
    PropertyEditor setBoolean(@PropertySet.BooleanProperty String name, boolean value);

    void commit();

    Box getBox(@PropertySet.BoxProperty String name);

    PropertyEditor setEventHandler(@PropertySet.EventProperty String name, int action, String data);

    PropertyEditor setString(@PropertySet.StringProperty String name, String value);

    PropertyEditor setInteger(@PropertySet.IntProperty String name, long value);

    PropertyEditor setEventHandler(@PropertySet.EventProperty String name, EventHandler eventHandler);
}

File: app/src/main/java/com/faendir/lightning_launcher/multitool/proxy/PropertyEditor.kt
package com.faendir.lightning_launcher.multitool.proxy

/**
 * @author lukas
 * @since 04.07.18
 */
interface PropertyEditor : Proxy {
    fun setBoolean(@PropertySet.BooleanProperty name: String, value: Boolean): PropertyEditor

    fun commit()

    fun getBox(@PropertySet.BoxProperty name: String): Box

    fun setEventHandler(@PropertySet.EventProperty name: String, action: Int, data: String?): PropertyEditor

    fun setString(@PropertySet.StringProperty name: String, value: String): PropertyEditor

    fun setInteger(@PropertySet.IntProperty name: String, value: Long): PropertyEditor

    fun setEventHandler(@PropertySet.EventProperty name: String, eventHandler: EventHandler): PropertyEditor
}
